client/client_pool/src/external_client.cpp

```diff
@@ -316,6 +317,8 @@ void ConcordClient::setDelayFlagForTest(bool delay) { ConcordClient::delayed_beh
 
 OperationResult ConcordClient::getRequestExecutionResult() { return clientRequestExecutionResult_; }
 
+std::string ConcordClient::messageSignature(bft::client::Msg& message) { return new_client_->signMessage(message); }
```

Comment:
We have already signed client requests. How is this different from the current approach?

Reply:
In https://github.com/vmware/concord-bft/blob/0bb1834b467342c5f449a3508975c2cc05ca8d28/reconfiguration/cmf/concord.cmf#L419
We can see the following declaration (snipped for easy reference)
Msg ReconfigurationRequest 1 {
  uint64 sender
  bytes signature
  oneof {
   <many reconfiguration commands>
  } command
  bytes additional_data
}

The above request is added with a jacket of ClientRequestMsg (or the above reconfiguraton request is a payload to ClientRequestMsg). Currently when creating ClientRequestMsg, one signature is added by default. But the dispatcher of Reconfiguration message gets the ReconfigurationRequest after the jacket is peeled off in ReplicaImp.  This dispatcher expects the signature of this payload in the signature field and thus we have to add this signature.
Dispatcher: https://github.com/vmware/concord-bft/blob/0bb1834b467342c5f449a3508975c2cc05ca8d28/reconfiguration/src/dispatcher.cpp#L38
So we have to add signature in the AddSenderAndSignature() function for ReconfigurationRequest when the actual client is decided by the client_pool.